zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/Learner.java

```diff
@@ -86,6 +89,7 @@
 
     protected Socket sock;
     protected MultipleAddresses leaderAddr;
+    protected AtomicBoolean sockBeingClosed = new AtomicBoolean(false);
```

Comment:
I think `socketClosed` would be a better name for this variable, because it's not only guarding the "closing" method.

Reply:
It is ONLY guarding the closing method. When it is set, the socket might not be closed for a long time, e.g. 30 seconds. But I don't want to name it "socketClosed" because it doesn't mean the socket is closed. Thoughts?